libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/tripsession/TripSessionComponent.kt

```diff
@@ -0,0 +1,96 @@
+package com.mapbox.navigation.dropin.component.tripsession
+
+import android.annotation.SuppressLint
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.repeatOnLifecycle
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.trip.session.TripSessionState
+import com.mapbox.navigation.ui.app.internal.Store
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
+import com.mapbox.navigation.ui.app.internal.tripsession.TripSessionStarterState
+import com.mapbox.navigation.ui.base.lifecycle.UIComponent
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.launch
+
+/**
+ * Component class that is responsible for starting and stopping [MapboxNavigation] trip session
+ * and replay trip session when [TripSessionStarterState] changes.
+ *
+ * @param lifecycle Parent view Lifecycle to which this component is attached to.
+ * @param store Store instance that holds NavigationState and TripSessionStarterState state
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+@SuppressLint("MissingPermission")
+internal class TripSessionComponent(
```

Comment:
Will you test this class with android auto?

Specifically, I want to be able to use this component delete this code here
https://github.com/mapbox/mapbox-navigation-android/blob/9f849f7f3051c85cfc6808474e8734ae1b482931/qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/car/MainCarSession.kt#L128-L135

Reply:
OK adding TripSessionComponent to this epic here https://github.com/mapbox/navigation-sdks/issues/1780